cosmos/providers/dbt/core/operators/local.py

```diff
@@ -365,6 +365,118 @@ def execute(self, context: Context):
         return result.output
 
 
+class DbtDocsS3LocalOperator(DbtDocsLocalOperator):
+    """
+    Executes `dbt docs generate` command and upload to S3 storage.
+
+    :param target_conn_id: S3's Airflow connection ID
+    :param bucket_name: S3's bucket name
+    :param folder_dir: This can be used to specify under which directory
+    the generated DBT documentation should be uploaded.
+    """
+
+    ui_color = "#FF9900"
+
+    def __init__(
+        self, target_conn_id: str, bucket_name: str, folder_dir: str = None, **kwargs
+    ) -> None:
+        from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+        self.target_conn_id = target_conn_id
+        self.bucket_name = bucket_name
+        self.folder_dir = folder_dir
+        super().__init__(**kwargs)
+        self.callback = self.upload_to_s3
+        self.s3_hook = S3Hook(
+            aws_conn_id=self.target_conn_id, extra_args={"ContentType": "text/html"}
+        )
+        if self.s3_hook.test_connection()[0] is not True:
+            logger.error("ERROR: Failed to connect S3")
+            raise AirflowException("Failed to connect S3")
+
+    def upload_to_s3(self, tmp_project_dir: str) -> None:
+        try:
+            target_dir = f"{tmp_project_dir}/target"
+
+            # iterate over the files in the target dir and upload them to S3
+            for dirpath, _, filenames in os.walk(target_dir):
+                for filename in filenames:
+                    if self.folder_dir is not None:
+                        key_path = f"{self.folder_dir}/{filename}"
+                    else:
+                        key_path = filename
+                    self.s3_hook.load_file(
+                        filename=f"{dirpath}/{filename}",
+                        bucket_name=self.bucket_name,
+                        key=key_path,
+                        replace=True,
+                    )
+        except ImportError:
+            logger.error("ERROR: the S3Hook isn't installed")
+        except Exception as exc:
+            logger.error("ERROR: " + str(exc))
+
+        return
```

Comment:
I think we can get rid of the try/except here since the import is now local, but I could be wrong. Do you mind testing that assumption? At the very least we should remove the last `except Exception as exc` and let the task just fail